goleft_functions.wdl

```diff
@@ -315,67 +343,51 @@ task report {
 	}
 
 	runtime {
+		disks: "local-disk " + reportDiskSize + " HDD"
 		docker: "python:3.8-slim"
-		preemptible: select_first([reportPreempt, 3])
-		memory: select_first([reportMemSize, 2]) + "G"
+		preemptible: reportPreempt
+		memory: reportMemSize + "G"
 	}
 }
 
 workflow goleft_functions {
 	input {
+		Boolean forceIndexcov = true
 		Array[File] inputBamsOrCrams
 		Array[File]? inputIndexes
 		File? refGenome
 		File? refGenomeIndex # currently unused
-
-		Boolean forceIndexcov = true
-
-		# runtime attributes with defaults
-		Int? covstatsAddlDisk
-		Int? covstatsMem
-		Int? covstatsPreempt
-		Int? indexcovAddlDisk
-		Int? indexcovMemory
-		Int? indexcovPrempt
-		Int? indexrefAddlDisk
-		Int? indexrefMem
-		Int? indexrefPreempt
-		Int? reportMem
-		Int? reportPreemptible
 	}
 
-	# Weird way to fallback if no indicies are defined, necessary due to how
-	# cloud localization works. There may another way to do this, however.
-	Array[String] wholeLottaNada = []
-
-	# doesn't work -- is there a WDL builtin that can print?
-	#if (length(allIndexes) != length(inputBamsOrCrams)) {
-			#echo "Warning: Not all files have an index. If all inputs are BAMs expect a slowdown in covstats."
-			#echo "In addition, indexcov will be skipped."
-	#}
+	Array[String] emptyArray = []
 
 	if(defined(refGenome)) {
 		call indexRefGenome { input: refGenome = refGenome }
 	}
 
 	scatter(oneBamOrCram in inputBamsOrCrams) {
 
-		Array[String] allOrNoIndexes = select_first([inputIndexes, wholeLottaNada])
+		Array[String] allOrNoIndexes = select_first([inputIndexes, emptyArray])
 
 		if (forceIndexcov || length(allOrNoIndexes) == length(inputBamsOrCrams)) {
 
 			String thisFilename = "${basename(oneBamOrCram)}"
 			String longerIfACram = sub(thisFilename, "\\.cram", "foobarbizbuzz")
```

Comment:
This workaround to determine the extension is perfectly functional, but doesn't sit well with me.

Reply:
WDL doesn't seem to have a better way other than creating an entire job to check this.  I think it might be best to leave it for now.